tests/client_tests/node-postgres/app.js

```diff
@@ -0,0 +1,109 @@
+
+const uuid = require('uuid/v4');
+const path = require('path');
+const expect = require('chai').expect;
+const Pool = require('pg').Pool;
+
+
+const pgClientPool = new Pool({
+    user: 'crate',
+    password: '',
+    host: process.argv.length === 4 ? process.argv[2] : 'localhost',
+    port: process.argv.length === 4 ? process.argv[3] : 5432
+})
+const dataFile = path.resolve(path.dirname(require.main.filename), 'resources/log_entries.csv');
+const testTableName = '"doc"."tmp_table_' + `${uuid()}`.substring(0, 16) + '"';
+
+
+function executeAsync(sql, callback) {
+    console.log(`Executing: ${sql}`);
+    pgClientPool.query(sql, (error, resultSet) => {
+        if (error) {
+            console.error(`The horror: ${error}`);
+            throw error;
+        }
+        callback(collect(resultSet));
+    });
+}
+
+
+async function execute(sql) {
+    console.log(`Executing: ${sql}`);
+    try {
+        await pgClientPool.query(sql);
+    } catch (error) {
+        console.error(`The horror: ${error}`);
+        throw error;
+    }
+}
+
+
+function collect(resultSet) {
+    let data = [];
+    data.colNames = resultSet.fields.map(f => f.name);
+    for (let i = 0; i < resultSet.rows.length; i++) {
+        let row = resultSet.rows[i];
+        let rowData = {};
+        for (let j = 0; j < data.colNames.length; j++) {
+            let colName = data.colNames[j];
+            rowData[colName] = row[colName];
+        }
+        data.push(rowData);
+    }
+    return data;
+}
+
+
+function logData(data) {
+    let colNames = data.colNames;
+    let str = `${data.length} results\n`;
+    for (let i = 0; i < data.length; i++) {
+        str += `Row ${i}:\n`;
+        for (let j = 0; j < colNames.length; j++) {
+            let colName = colNames[j];
+            let colValue = data[i][colName];
+            str += `  ${colName} = ${colValue}\n`;
+        }
+    }
+    console.log(str);
+}
+
+
+// The tests
+(async () => {
+    await execute(
+        `CREATE TABLE ${testTableName} (` +
+        '        log_time timestamp NOT NULL,' +
+        '        client_ip ip NOT NULL,' +
+        '        request string NOT NULL,' +
+        '        status_code short NOT NULL,' +
+        '        object_size long NOT NULL);'
+    );
+    await execute(`COPY ${testTableName} FROM 'file://${dataFile}';`);
```

Comment:
Maybe some INSERT case would also be nice to have?

Reply:
I was thinking in having a separate PR adding adding code with two separate calls, one to copy, the other to insert, and maybe deal with 20K rows. For high volumes of ingest it may be faster to dump into a tmp file, then copy, vs. batched inserts. But I can just add a few inserts to the existing code and be done. Should I just add the inserts?

Reply:
I'd just add the inserts. We don't have to turn this into a benchmark, we just want to ensure that basic functionality works.